Dialogs in java fx

velissarious opened the pull request.

velissarious committed:
add FXML and controller for login

Initial work on FXML form and controller that will eventually replace the existing non-JavaFX login screen.
add LoginViewWindow and a new MangeHPCWorkflow


add directory chooser


add some form validation

Check if fields are empty, check if directory is used.
improve form validation


fix now it opens the job detail view


enable snakemake output tab for macro but rename to error output


feature: remember previous login settings on the login view

The previous settings will be stored using prefService in order to be available upon restarting the program in order to populate the form fields.
add: new dialog to display exceptions, initial progress dialog

Add a new dialog to display exceptions in SimpleDialog.
Add initial FXML view and Controller for the new JavaFX progress dialog.
fix: check file lock properly on login, replace old progress dialogs

Fix a bug introduced in the previous commit, checking if the working directory was already in use did not work, now it is fixed.

Improved progress view and controller, replaced some of the old progress dialogs with the new one.
fix: replaced more progress dialogs with new

Initial work to convert the executeWSCallAsync method's progress dialogs to the new one.

velissarious committed:
clean code


clean code

velissarious committed:
fix executeWSCallAsync

Trying to fix executeWSCallAsync in BenchmarkSPIMControl by adding the missing necessary functionality to the new implementation of ProgressNotifier (ProgresDialogViewWindow).

velissarious committed:
fix ProgressDialogViewWindow

remove redundant methods from ProgressDialogViewWindow

fix: disable x window button on progress dialog

Since JavaFX does not have a method to disable all window decoration buttons without also losing the window handle, I chose to hide resize and disable the functionality of the x button (which remains visible).
fix: reset progress after finishing a task


add proper item sub progress in ScrollPane

Each sub-item of a progress dialog will have its own progress bar in a GridPane inside a ScrollPane. Visible only if the ToggleButton labeled "Show Details" is pressed.
remove useless method "resetProgress"

The overall progress in a progress dialog is never reset.
fix: set item progress to 100% when done

Progress dialog improvement.
Removed a useless include.

velissarious committed:
add status bar in Macro progress view

Since there is no status bar in JavaFX a Label was added instead in the HBox (at the bottom, alight right).

velissarious committed:
enable jobCanBeDownloaded check

velissarious committed:
rebase and fix conflicts


rename BenchmarkJobManager to HPCWorkflowManager

velissarious committed:
remove SimpleDialog class and include swing-javafx-ui SimpleDialog


remove JavaFXRoutines and use swing-javafx-ui instead

Add swing-javafx-ui dependency in pom.
Replace all relevant method calls to swing-javafx-ui method calls.

velissarious committed:
change login and HPCWorkflow work with JavaFX only

velissarious committed:
change jobDetails to use JavaFX fully

Modality does not work well enough yet.
fix: resizeable JobDetail and HPCWorkflow windows

Made JobDetail and HPCWorkflow resizable again.
disable modality of JobDetail, user can see many detail views of jobs 

Modality of progress dialogs still does not work as expected.

velissarious committed:
fully convert "Create Job" dialog to JavaFX

velissarious committed:
fix MacroTaskProgressViewController

It was broken after the renaming of the packages etc...

Now it is fixed.

velissarious committed:
rename Benchmark.fxml to HPCWorkflow.fxml

This file was missed during refactoring and had to be renamed.

Also made a small modification to the TreeView in order to display only the columns that exist and resize them when the table is resized.

This is to conserve screen space and reduce user's effort to resize the columns when the window is resized.

velissarious committed:
disable demo option from create job dialog if macro workflow is selected

velissarious committed:
changed the number of nodes TextField to a Spinner in create job dialog

This will prevent incorrect input (for example characters) by the user.

It is also more user friendly as it can be used with the mouse as well as the keyboard.

velissarious committed:
add always on top property to progress dialogs


fix: remove swing, add: display details button only if details exist

velissarious committed:
convert SPIM related UI to JavaFX

This requires more testing.

velissarious committed:
removed unused code JobOutputView and RunView

There is some unused swing code that also had a test, both are removed with no impact on the functionality of the program.

kozusznik commented on hpc-workflow-manager-client/src/main/java/cz/it4i/fiji/hpc_workflow/commands/HPCWorkflowParametersImpl.java:
```diff
@@ -1,27 +1,34 @@
 import cz.it4i.fiji.hpc_workflow.core.HPCWorkflowParameters;
 
-class HPCWorkflowParametersImpl implements HPCWorkflowParameters{
+public class HPCWorkflowParametersImpl implements HPCWorkflowParameters, Serializable {
```
Add static field identifing version for serialization.

velissarious replied:
Added a serialVersionUID.

kozusznik commented on hpc-workflow-manager-client/src/main/java/cz/it4i/fiji/hpc_workflow/commands/NewManageHPCWorkflow.java:
```diff
@@ -0,0 +1,51 @@
+@Plugin(type = Command.class, headless = false, priority = Priority.HIGH,
+	menuPath = "Plugins>" + Constants.MENU_ITEM_NAME + ">" +
+		Constants.SUBMENU_ITEM_NAME)
+public class NewManageHPCWorkflow implements Command {
```
Prefix New seems to me unnecessary.

velissarious replied:
I removed the prefix "New".

kozusznik commented on hpc-workflow-manager-heappe-adapter/src/main/java/cz/it4i/fiji/haas/ui/SwingAndJavaFXLinker.java:
```diff
@@ -15,18 +17,21 @@
  */
 
-public class JFXPanel<T extends Parent> extends javafx.embed.swing.JFXPanel {
+public class SwingAndJavaFXLinker<T extends Parent> extends javafx.embed.swing.JFXPanel {
```
Should be removed - it is in library.

kozusznik reviewed (commented):
Also, consider how many commits should be contained in the master after integration and squash it.

velissarious committed:
cleanup and minor improvements to dialogs

Made some changes to remove two warnings.

Changed exception dialogs to JavaFX

simplify exception dialogs


replace information dialog with new warning dialog


rename NewManageHPCWorkflow to ManageHPCWorkflow


add serialVersionUID to Serializable class HPCWorkflowParametersImpl


update pom

kozusznik committed:
code: Remove FXFrame,FXFrameNative - use them from swing-javafx-ui

velissarious committed:
add a reminder not to change this line

Added a reminder not to change the line because SonarLint is highlighting it and someone might try to change it again.

velissarious committed:
fix: resize progress dialog when the details button is pressed

velissarious committed:
change column resize policy to constrained-resize except progress

All column resize policy is set to constrained-resize except the two progress screens for Macro and SPIM.

velissarious committed:
fix RunRemoteFilesView test to run correctly

velissarious committed:
fix bug in MacroTask

This bug prevented displaying the progress of a running job reliably as there was an exception that occurred silently. This was most apparent when running job with many nodes. Jobs running on many nodes should now work reliably.

velissarious committed:
unit test for MacroTask


add more tests for MacroTask

velissarious committed:
fix: error output is now working for macro

The error output tab was labeled Snakemake output in the past and was therefore disabled for Macro. Later it was re-titled as "Error" tab for the macro but the actual functionality was not re-enabled. Now it is enabled and works. The error tab will be enabled when there are errors to display. Also renamed the "snakemakeOutputControl" to "logViewControl" as it is now used for displaying errors for the Macro Workflow type as well.

kozusznik commented on hpc-workflow-manager-client/src/main/java/cz/it4i/fiji/hpc_workflow/ui/MacroTaskProgressViewController.java:
```diff
@@ -254,6 +251,13 @@ private int setTaskProgressOrDescriptionFromElements(int nodeId,
 				nodeTaskToDescription.get(nodeId).put(taskIdForNode, description);
 			}
+			catch (Exception exc) {
+				if (isFirstTime) {
```
Consider whether does make sense to set the flag isFirstTime to true when parsing does not throw error.

velissarious replied:
I replaced this code, now parsing will stop, the progress will stop updating and an error displaying the unknown catch exception will be displayed.

kozusznik commented on hpc-workflow-manager-client/src/main/java/cz/it4i/fiji/hpc_workflow/ui/ProgressDialogViewController.java:
```diff
@@ -0,0 +1,120 @@
+			
+			// Increase the stage size to fit the details:
+			Stage stage = (Stage) taskDescriptionLabel.getScene().getWindow();
+			double originalHeight = 150.0;
```
Consider using constant(may be local) instead of magic values.

velissarious replied:
I used local constants instead.

kozusznik commented on hpc-workflow-manager-client/src/test/java/cz/it4i/fiji/haas/MacroTaskTest.java:
```diff
@@ -0,0 +1,65 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
```
parent pom scijava defines JUnit version 4.12 in dependency management.

velissarious replied:
I removed JUnit 5 from my eclipse. Added JUnit in pom.

velissarious committed:
make review changes

Add JUnit 4 to POM.

Replace "magic numbers" with local constants.

Catch catastrophic errors in Macro progress file parsing, stop the executor and display dialog with the caught exception.
removed test code

velissarious committed:
fix MacroTask now prevents progress greater than 100 and less than 0

velissarious committed:
remove unused code in MacroWrapper and add new scatter method


fix parScatterEqually


fix: remove unused code

velissarious committed:
remove unnecessary showAndWait and use SimpleDialog for warning dialog

velissarious committed:
remove unused code

WindowClosableAdapter is not used anymore after conversion to JavaFX and therefore it should be removed.
